aries_cloudagent/messaging/models/base_record.py

```diff
@@ -0,0 +1,367 @@
+"""Classes for BaseStorage-based record management."""
+
+import json
+import sys
+import uuid
+
+from datetime import datetime
+from typing import Any, Mapping, Sequence, Union
+
+from marshmallow import fields
+
+from ...cache.base import BaseCache
+from ...config.injection_context import InjectionContext
+from ...storage.base import BaseStorage
+from ...storage.record import StorageRecord
+
+from .base import BaseModel, BaseModelSchema
+from ..responder import BaseResponder
+from ..util import datetime_to_str, time_now
+
+
+class BaseRecord(BaseModel):
+    """Represents a single storage record."""
+
+    class Meta:
+        """BaseRecord metadata."""
+
+    RECORD_ID_NAME = "record_id"
```

Comment:
This is a nit - but do you think we could just call this "id"? Is there a reason to name it explicitly?

Reply:
It's mainly for compatibility with the existing stored records where it's named connection_id, credential_exchange_id, etc. The ID isn't stored that way but it would still be a lot of references to update.

Reply:
Also `id` is a python keyword..

Reply:
I think `id()` is a built-in function but it should be safe to use as a property on an object.

Right, forgot about the current stuff. However - arguably, id makes more sense in a vacuum and we're still early enough that we can introduce breaking changes.

What do you think about defaulting to using `id` and allowing it to be overridden?